lib/read_parsers.cc

```diff
@@ -58,8 +58,20 @@ void SeqAnParser::imprint_next_read(Read &the_read)
         ret = seqan::readRecord(the_read.name, the_read.sequence,
                                 the_read.quality, _private->stream);
         if (ret == 0) {
+            if (_num_reads == 0 && the_read.quality.length() != 0) {
+                _have_qualities = true;
+            }
             _num_reads++;
         }
+        if (the_read.sequence.length() == 0) {
```

Comment:
Everything should be outside the spinlocks as much as possible. Especially throwing an exception. Looks like the `_num_reads` increment could also be moved after the spin lock is released.